service/history/workflow/task_refresher.go

```diff
@@ -389,12 +389,13 @@ func (r *TaskRefresherImpl) refreshTasksForActivity(
 		}
 
 		if CompareVersionedTransition(minVersionedTransition, EmptyVersionedTransition) == 0 { // Full refresh
-			// clear activity timer task mask for later activity timer task re-generation
-			activityInfo.TimerTaskStatus = TimerTaskStatusNone
 
 			// need to update activity timer task mask for which task is generated
 			if err := mutableState.UpdateActivity(
```

Comment:
cc @xwduan is [adding](https://github.com/temporalio/temporal/pull/6816) a dedicated method for updating timer task status since mutable state needs to know if the updated field supposed to be replicated.